### PR TITLE
docs(TRACT): Add thread control docs

### PR DESCRIPTION
**Pull request by 1a8e**
- Talk about TRACT TC in multi chat tool scenarios

https://hello-haptik.atlassian.net/browse/AGCT-870

### CODE REVIEW

File: docs/external-chat-tool/thread-control.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+title: Thread control
+---
+
+> Note: Please refer to [actors](https://docs.haptik.ai/external-chat-tool/#actors) for the values of the abbreviations used.
+
+## What is thread control?
+
+Thread control is a way of assigning ownership of a chat conversation (a.k.a. "thread") to a chat tool capable of responding to a CEU.
+
+The thread control protocol allows for multiple ECTs to be able to respond to a given chat form a CEU, in conjunction with one another and the Haptik internal tools: Smart Agent Chat (SAC) and the Intelligent Virtual Assistant (IVA).
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:11]:
```suggestion
The thread control protocol allows for multiple ECTs to be able to respond to a given chat from a CEU, in conjunction with one another and the Haptik internal tools: Smart Agent Chat (SAC) and the Intelligent Virtual Assistant (IVA).
```

File: docs/external-chat-tool/thread-control.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+The chat tool that has thread control at a given instance is the only chat tool allowed to respond to the CEU at at instance.
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:13]:
```suggestion
The chat tool that has thread control at a given instance is the only chat tool allowed to respond to the CEU at instance.
```

Comment (medhaanand) [reply docs/external-chat-tool/thread-control.md:13]: Not sure what is the meaning of this sentence.

Comment (medhaanand) [reply docs/external-chat-tool/thread-control.md:13]: Also have the next sentence in continuation, remove the extra space.

File: docs/external-chat-tool/thread-control.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+If another chat tool wishes to respond to the CEU, they must first be given thread control via a thread control operation.
+
+## Why would you need TC / should you be using TC?
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:17]: TC is thread control I assume. Please specify it somewhere.

File: docs/external-chat-tool/thread-control.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+Most of our clients that use TRACT, utilise a Haptik IVA along with a single ECT. This typical use-case manages thread control implicitly and you do not need to manually trigger TC operations. 
+
+Explicit thread control features relevant to you in the following use-cases:
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:21]:
```suggestion
Explicit thread control feature is relevant to you in the following use-cases:
```

File: docs/external-chat-tool/thread-control.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+- `is_primary`: Primary receivers
+- `active` : Active receiver
+
+### What is a default receiver?
+
+For a given client, the default receiver configured gets control of the conversation when a CEU first sends a message.
+This is applicable for:
+
+- a new user's chat
+- a chat for an old user once either of the following has taken place
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:46]: Is it for a new chat for an old user?

File: docs/external-chat-tool/thread-control.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  - TC has been reset for said chat
+  - said chat has been marked complete
+
+If an explicit receiver is configured as the default receiver, they are always given the TC in the above two cases. You can make either an ECT or one of the internal tools as the default receiver.
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:50]:
```suggestion
If an explicit receiver is configured as the default receiver, they are always given the TC in the above two cases. You can either make an ECT or one of the internal tools as the default receiver.
```

File: docs/external-chat-tool/thread-control.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+If an explicit default receiver is not set, we use the following priority ladder to decide the default:
+
+- If an IVA is configured for this client, the IVA is made default.
+- If an IVA is not configured but SAC is enabled, the SAC is made default.
+- If neither of IVA and SAC are configured, but only one ECT is configured, said ECT is made default.
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:56]:
```suggestion
- If neither of IVA and SAC are configured, but only one ECT is configured, the ECT is made default.
```

File: docs/external-chat-tool/thread-control.md (hunk 1 above)
Comment (medhaanand) [reply docs/external-chat-tool/thread-control.md:11]: Also how is IVA a Haptik internal tool?

File: docs/external-chat-tool/thread-control.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+This simply is the current thread owner, currently allowed to actively respond to the CEU.
+
+### What thread control operation can I perform?
+
+Unless negotiated otherwise, as an ECT **you can only pass thread control to one of the primary or the default receiver**.
+
+Hence,
+
+- Requesting of thread control is currently not supported.
+- Taking thread control is currently not supported, unless you are a primary.
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:81]:
```suggestion
- Taking thread control is currently not supported, unless you are primary.
```

File: docs/external-chat-tool/thread-control.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+> TRACT only allows TC operations from the active or primary receivers as of now.
+
+To perform a thread control operation request please use the `thread_control/request` API. Please refer the detailed API documentation [here](https://docs.haptik.ai/external-chat-tool/thread-control-APIs#api-thread-management).
+
+### Summary table for receivers
+
+| Type    | Can respond to CEU? | Can take thread control from others? | Can give thread control to others? | Gets first message from new CEU chat? | Who can this be? |
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:89]:
```suggestion
| Type    | Can respond to CEU? | Can take thread control from others? | Can give thread control to others? | Gets first message from new CEU chat | Who can this be? |
```

Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:89]: Why?

File: docs/external-chat-tool/thread-control.md (hunk 8 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:81]: Can be >1 primary.

File: docs/external-chat-tool/thread-control.md (hunk 7 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:56]: Why?

File: docs/external-chat-tool/thread-control.md (hunk 6 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:50]: Why?

File: docs/external-chat-tool/thread-control.md (hunk 5 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:46]: Both na... what's the confusion?

File: docs/external-chat-tool/thread-control.md (hunk 8 above)
Comment (medhaanand) [reply docs/external-chat-tool/thread-control.md:92]: then let us call it primary receiver.

File: docs/external-chat-tool/thread-control.md (hunk 7 above)
Comment (medhaanand) [reply docs/external-chat-tool/thread-control.md:64]: "said" is not giving the correct impression here. Can we say this instead?

- If neither of IVA and SAC are configured, but only one ECT is configured, then this ECT is made default.

File: docs/external-chat-tool/thread-control.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+### What is a default receiver?
+
+For a given client, the default receiver configured gets control of the conversation when a CEU first sends a message.
+This is applicable for:
+
+- a new user's chat
+- a chat for an old user once either of the following has taken place
+    - TC has been reset for that chat
+    - That chat has been marked complete
```
Comment (medhaanand) [inline docs/external-chat-tool/thread-control.md:55]:
```suggestion
    - a chat that has been marked complete
```

File: docs/external-chat-tool/thread-control.md (hunk 5 above)
Comment (medhaanand) [reply docs/external-chat-tool/thread-control.md:53]: Alright two things here -
- What is the meaning of "once either of the following has taken place"
- when you say "a chat for an old user" do you mean a new chat for an old user or an old chat reopening for an old user

File: docs/external-chat-tool/thread-control.md (hunk 3 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:21]: +1

File: docs/external-chat-tool/thread-control.md (hunk 4 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:26]: +1

File: docs/external-chat-tool/thread-control.md (hunk 7 above)
Comment (1a8e) [reply docs/external-chat-tool/thread-control.md:64]: +1

**Review by medhaanand — APPROVED** [review]

**Pull request merged**